File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if (speed == 0) return 0;
        if (speed >= 1 && speed <= 4) return 1;
        if (speed >= 5 && speed <= 8) return 0.9;
        if (speed == 9) return 0.8;
        if (speed == 10) return 0.77;
        throw new NotImplementedException("Please implement the (static) AssemblyLine.SuccessRate() method");
    }
    
    public static double ProductionRatePerHour(int speed)
        => SuccessRate(speed) * 221 * speed; 

    public static int WorkingItemsPerMinute(int speed)
        => (int)(ProductionRatePerHour(speed) / 60);
}

File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {
    public double productionRatePerHour(int speed) {
        double res = speed * 221;
        if (speed == 0)
            res = 0;
        else if (speed > 4 && speed < 9)
            res *= 0.9;
        else if (speed == 9)
            res *= 0.8;
        else if (speed == 10)
            res *= 0.77;
        return res;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed) / 60);
    }
}
